docs/guides/authjs-migration.mdx

```diff
@@ -0,0 +1,390 @@
+- build a [fully custom UI using Clerk's API](/docs/custom-flows/overview), leveraging Clerk's React hooks such as [`useSignIn()`](/docs/references/react/use-sign-in) and [`useSignUp()`](/docs/references/react/use-sign-up)
+
+
+### Protect your app
+
+With Clerk, you can control access to your application in a few different ways. One way is to use Clerk's Middleware to protect your entire application, or specific routes. Another way is to use Clerk's components to conditionally render UI based on the user's authentication state. You can hide or show UI based on whether the user is signed in or not.
+
+#### Control access to your app with Clerk Middleware
+
+You will need to remove the Auth.js Middleware from your application, and replace it with the Clerk's Middleware helper, `authMiddleware()`. 
+
+Auth.js's middleware always rejects unauthorized requests. You may have additionally configured the Next.js middleware config to protect specific private routes. You will need to make note of this configuration so you can recreate it.
+
+Clerk's Middleware gives you fine-grained control over handling the authenticated state and will, by default, run for your entire application.
+
+The example below is a basic configuration that requires authentication for all routes. You will need to edit this to allow specific routes to be public, as needed. Read the [`authMiddleware()`](/docs/references/nextjs/auth-middleware) documentation to learn more about how you can configure your Middleware.
+
+```tsx filename="middleware.ts"
+import { authMiddleware } from "@clerk/nextjs";
+ 
+// This example protects all routes including api/trpc routes
+// Please edit this to allow other routes to be public as needed.
+// See https://clerk.com/docs/references/nextjs/auth-middleware for more information about configuring your Middleware
+export default authMiddleware({});
+ 
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+#### Control access to your app with Clerk's components
+
+To conditionally render UI when the user is signed in, wrap it with [`<SignedIn>`](/docs/components/control/signed-in).
+
+To conditionally render UI when the user is _not_ signed in, wrap it with [`<SignedOut>`](/docs/components/control/signed-out).
+
+```tsx filename="app/page.tsx"
+import { SignedIn, SignedOut } from "@clerk/nextjs";
+ 
+export default function Home() {
+  return (
+    <div>
+      <SignedOut>
+        <p>This content is public. Only signed out users can see this.</p>
+      </SignedOut>
+      <SignedIn>
+        <p>This content is private. Only signed in users can see this.</p>
+      </SignedIn>
+    </div>
+  )
+}
+```
+
+### Read user and session data
+
+#### Server side
+
+Replace any Auth.js `getServerSession(req, res, authOptions)` with Clerk's helpers.
+
+Let's see what your code might look like with Auth.js's `getServerSession()`:
+
+```tsx filename="app/page.tsx"
```

Comment:
We have App Router & Pages Router examples all thru out this guide. We should stay consistent and show here what the `getServerSession()` usage would like in their Pages Router app

Reply:
This is leftover and needs deleting. The examples for both are below in Tabs.

Reply:
This is the Auth.js code. Based on your previous edits I think we want to delete this.

Reply:
omg you know what, I just thought of something - lennart uses a `diff` syntax on the code blocks to create diffs ([example](https://github.com/clerk/clerk-docs/pull/605/files#diff-3e9e3463657da8ff372ec5ca21d680029e440e06db876ea3b9754225316eed5eR177)).

we could use that here, so show auth.js code you would remove , and clerk code that you would add.
what do you think??

Reply:
We could. My feeling is that your first instinct to not include the Auth.js code is correct. Chances are good that the implementation is much more complex than the sample, so I think its limited value.

Reply:
okay let's remove it!